lib/capabilities.go

```diff
@@ -78,3 +91,31 @@ func CapAwayNotify(caps *CapManager) {
 		},
 	)
 }
+
+func CapServerTime(caps *CapManager) {
+	name := "server-time"
+	caps.Supported[name] = ""
+
+	caps.FnsInitListener[name] = func(listener *Listener) {
+		listener.TagsEnabled = true
+	}
+
+	caps.FnsMessageToClient = append(
+		caps.FnsMessageToClient,
+		func(listener *Listener, message *ircmsg.IrcMessage) bool {
+			if !listener.IsCapEnabled(name) {
+				return false
+			}
+
+			_, exists := message.Tags["time"]
+			if !exists {
+				message.Tags["time"] = ircmsg.TagValue{
```

Comment:
I was going to ask about the TagValue thing as it's awkward and redundant as it is. Surely just a string value is needed, and checking if a value is set is a case of `if msg.tags["foo"] == ""`? Values are strings anyway.